src/types.rs

```diff
@@ -2,13 +2,98 @@
 pub type BlockHeight = u64;
 
 /// Configuration struct for NEAR Lake Framework
+/// NB! Consider using [`LakeConfigBuilder`]
+/// Building the `LakeConfig` example:
+/// ```
+/// use near_lake_framework::LakeConfigBuilder;
+/// # async fn main() {
+///    let config = LakeConfigBuilder::default()
+///        .s3_bucket_name("near-lake-data-testnet")
+///        .start_block_height(82422587)
+///        .build()
+///        .expect("Failed to build LakeConfig");
+/// # }
+/// ```
+#[derive(Default, Builder, Debug)]
 pub struct LakeConfig {
     /// AWS S3 compatible custom endpoint
+    ///
+    /// In case you want to run your own [near-lake](https://github.com/near/near-lake) instance and store data in some S3 compatible storage ([Minio](https://min.io/) or [Localstack](https://localstack.cloud/) as example)
+    ///
+    /// You can owerride default S3 API endpoint by using `s3_endpoint` option or [LakeConfigBuilder::with_custom_endpoint]"]
+    #[builder(setter(custom, strip_option, name = "with_custom_endpoint"), default)]
```

Comment:
I would suggest to stick to either have `with_` prefix everywhere or nowhere since otherwise it looks odd.